### PR TITLE
Move load tablets to StorageEngine

### PR DESCRIPTION
**Pull request by yiguolei**
1. move load tablets logic from data dir to storageengine, data dir only do dir and meta management work.
2. change rootpathinfo to datadirinfo.

### CODE REVIEW

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -1819,11 +1819,11 @@ void* TaskWorkerPool::_report_disk_state_worker_thread_callback(void* arg_this)
             continue;
         }
 #endif
-        vector<RootPathInfo> root_paths_info;
-        worker_pool_this->_env->olap_engine()->get_all_root_path_info(&root_paths_info);
+        vector<DataDirInfo> data_dir_infos;
+        worker_pool_this->_env->olap_engine()->get_all_root_path_info(&data_dir_infos);
```
Comment (kangpinghuang) [inline be/src/agent/task_worker_pool.cpp]: get_all_root_path_info should be renamed?

File: be/src/olap/storage_engine.cpp [hunk 2]
```diff
@@ -966,8 +915,8 @@ OLAPStatus StorageEngine::start_trash_sweep(double* usage) {
     const uint32_t snapshot_expire = config::snapshot_expire_time_sec;
     const uint32_t trash_expire = config::trash_file_expire_time_sec;
     const double guard_space = config::disk_capacity_insufficient_percentage / 100.0;
-    std::vector<RootPathInfo> root_paths_info;
-    res = get_all_root_path_info(&root_paths_info);
+    std::vector<DataDirInfo> data_dir_infos;
+    res = get_all_root_path_info(&data_dir_infos);
```
Comment (kangpinghuang) [inline be/src/olap/storage_engine.cpp]: get_all_root_path_info should be renamed?

**Review by kangpinghuang — COMMENTED** [review]
LGTM

File: be/src/agent/task_worker_pool.cpp (hunk 1 above)
Comment (yiguolei) [reply be/src/agent/task_worker_pool.cpp]: I will modify it.

**Review by kangpinghuang — APPROVED** [review]

**Pull request merged**